ethcore/light/src/on_demand/request.rs

```diff
@@ -650,9 +651,10 @@ impl From<::rlp::DecoderError> for Error {
 	}
 }
 
-impl From<Box<TrieError>> for Error {
-	fn from(err: Box<TrieError>) -> Self {
-		Error::Trie(*err)
+impl<T> From<Box<TrieError<T>>> for Error {
+	fn from(err: Box<TrieError<T>>) -> Self {
+		// Error::Trie(*err)
+		Error::Trie(TrieError::InvalidStateRoot(<KeccakHasher as Hasher>::Out::new())) // REVIEW: how do I fix this without making `Error` generic also?
```

Comment:
maybe `TrieError::InvalidStateRoot(String)` and implement conversion to it for every `Hasher::Out + Debug` ?

Reply:
Not sure I follow. `TrieError::InvalidStateRoot` here is plain wrong and only temporary to make the code compile. I honestly don't know how to deal with the generic `TrieError<T>` in higher level crates that don't use generic errors (and error_chain – which cannot have generic errors as foreign link). @rphmeier suggested in PM that we make a type alias for `TrieError` but I am not entirely sure how that's going to work.

Reply:
> Not sure I follow. TrieError::InvalidStateRoot here is plain wrong and only temporary to make the code compile

I see

>  I honestly don't know how to deal with the generic TrieError<T> in higher level crates that don't use generic errors (and error_chain – which cannot have generic errors as foreign link). @rphmeier suggested in PM that we make a type alias for TrieError but I am not entirely sure how that's going to work.

What about not having TrieError generic at all. Just having one of the variants of it wrap `Box<io::Error>`?

Reply:
we tried something pretty close. the issue is we need `Clone` + `Debug` for `TrieError` further up in the hierarchy and `Box<error::Error>` isn't clone

Reply:
> From<Box<TrieError<T>>> for Error

Using the generic vesion of `TrieError` is the issue here. The _actual_ conversion we want to support is `From<Box<TrieError<H256>>>`